lang/rust/avro/src/schema.rs

```diff
@@ -1518,7 +1528,7 @@ impl Parser {
         }
 
         let schema = Schema::Enum(EnumSchema {
-            name,
+            name: fully_qualified_name.clone(),
```

Comment:
Wouldn't it have been sufficient to ; keeping 
`let fully_qualified_name = name.fully_qualified_name(enclosing_namespace);`
without modifying Name::parse method ?

Reply:
It's not enough because [this test](https://github.com/apache/avro/pull/2405/files#diff-b702ef60e00210dede234ba524d4ce1d4ff14ef199191ec46a7f4ba8a6bd211eR4944) doesn't pass.